entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/service/SolrService.java

```diff
@@ -155,19 +163,17 @@ public void storeMultipleEntities(final List<SolrEntity<? extends Entity>> solrE
    * @return SolrEntity instance
    * @throws SolrServiceException if error occurs while executing query
    */
-  public <T extends Entity, U extends SolrEntity<T>> U searchById(
-      Class<U> classType, String entityId) throws SolrServiceException {
+  public SolrEntity<? extends Entity> searchById(String entityId) throws SolrServiceException {
```

Comment:
yes, indeed, the previous implementation works when we are retrieving only one Enntity but it doesn't work for the general search when the return list may return a mixed list of entities. I updated this method to use the same mapping to SolrEntities as the general search in order to avoid code dupplication, There is not much a difference if the concrete class is used for casting or if it is sent as method parameter, it has to be used anyway.